contrib/platform/src/com/sun/jna/platform/win32/Shell32.java

```diff
@@ -376,5 +377,30 @@
      * @see <a href="https://msdn.microsoft.com/en-us/library/ms648069(VS.85).aspx">MSDN</a>
      */
     int ExtractIconEx(String lpszFile, int nIconIndex, HICON[] phiconLarge, HICON[] phiconSmall, int nIcons);
+
+    /**
+     * Retrieves the application-defined, explicit Application User Model ID (AppUserModelID) for the current process.
+     * 
+     * @param ppszAppID
+     *            Type: PWSTR*<br>
+     *            A pointer that receives the address of the AppUserModelID assigned to the process. The caller is responsible for freeing this string with {@link Ole32#CoTaskMemFree} when it is no longer needed.
+     * @return Type: HRESULT<br>
+     *         If this function succeeds, it returns S_OK. Otherwise, it returns an HRESULT error code.
+     * @see <a href="https://msdn.microsoft.com/en-us/library/windows/desktop/dd378419(v=vs.85).aspx">MSDN</a>
+     */
+    HRESULT GetCurrentProcessExplicitAppUserModelID(PointerByReference ppszAppID);
+
+    /**
+     * Specifies a unique application-defined Application User Model ID (AppUserModelID) that identifies the current process to the taskbar. This identifier allows an application to group its associated processes and windows under a single taskbar button.
+     * 
+     * @param appID
+     *            Type: PCWSTR<br>
```

Comment:
I just copied the `Type: XXX` docs directly from MSDN. So you want me to remove those lines?